contracts/src/systems/hyperstructure/contracts.cairo

```diff
@@ -345,6 +347,10 @@ mod hyperstructure_systems {
             let winner_address = starknet::get_caller_address();
             world.emit_event(@GameEnded { winner_address, timestamp: starknet::get_block_timestamp() });
 
+            let mut leaderboard: Leaderboard = world.read_model(WORLD_CONFIG_ID);
+            leaderboard.registration_end_timestamp = starknet::get_block_timestamp() + LEADERBOARD_REGISTRATION_PERIOD;
```

Comment:
do we need to store registration end, if we already have GameEnded timestamp and LEADERBOARD_REGISTRATION_PERIOD?

Reply:
the `GameEnded` is an event being emitted, there's no way to get it from the contracts. We would have to switch it to a model then

Reply:
ok